### ISSUE TITLE
Number format integer translated to double

### ISSUE THREAD

**Issue by julznygit** [RicoSuter/NSwag#2880]
Hi,

I'm using NSwagStudio to translate YAML to C# Client, and found that the number format integer is still translated to double:

Here is the YAML Spec:

TotalRecords: 
type: number
format: Integer

I have downloaded the latest version of NSwagStudio. Can you please advise if this is a bug or it needs to be configured somewhere?

Thanks

Julius

**Comment by RicoSuter** [comment]
This is essentially a wrong spec, ie the type must be "integer"
https://github.com/RicoSuter/NJsonSchema/blob/master/src/NJsonSchema/JsonObjectType.cs#L32

Not sure whether we should handle this "wrong" spec in your expected way: 
https://github.com/RicoSuter/NJsonSchema/blob/master/src/NJsonSchema.CodeGeneration.CSharp/CSharpTypeResolver.cs#L97

and here
https://github.com/RicoSuter/NJsonSchema/blob/master/src/NJsonSchema.CodeGeneration.TypeScript/TypeScriptTypeResolver.cs#L106

**Comment by julznygit** [comment]
Hi Rico

 

Thanks for the clarification. I will ask the provider to change their YAML spec. If they don’t want to change the spec, is there any recommended workarounds?

 

Thanks

 

Julius

**Comment by RicoSuter** [comment]
> If they don’t want to change the spec, is there any recommended workarounds?

Preprocess the spec so that it fits the generator expectations...

**Comment by sherlock1982** [comment]
Yes according to the [specification](https://spec.openapis.org/oas/v3.1.0.html#data-types)

    type: number
    format: integer

Is really invalid. Though specification authors added additional registry of pairs that are optional to support but exist in the wild. https://spec.openapis.org/registry/format/

We hit it with `number/int32` for example. NSwag could have supported these optional pairs.
